resources/js/features/games/components/GameMobileHeader/GameMobileBannerImage/GameMobileBannerImage.tsx

```diff
@@ -0,0 +1,113 @@
+import { type FC, useRef, useState } from 'react';
+
+import { usePageProps } from '@/common/hooks/usePageProps';
+import { cn } from '@/common/utils/cn';
+
+export const GameMobileBannerImage: FC = () => {
+  const { game } = usePageProps<App.Platform.Data.GameShowPageProps>();
+
+  const [isImageLoaded, setIsImageLoaded] = useState(false);
+  const imageRef = useRef<HTMLImageElement>(null);
+
+  const handleImageRef = (element: HTMLImageElement | null) => {
+    imageRef.current = element;
+
+    if (element?.complete) {
+      setIsImageLoaded(true);
+    }
+  };
+
+  const isNintendoDS = game.system?.id === 18;
+
+  return (
+    <>
+      <div className="absolute inset-0 overflow-hidden">
+        {game.banner?.mobileSmWebp ? (
+          <>
+            {/*
+             * Blurred placeholder - loads instantly.
+             * This is critical for LCP / Core Web Vitals.
+             *
+             * @see Game::registerMediaCollections
+             */}
+            {game.banner.mobilePlaceholder ? (
+              <img
+                src={game.banner.mobilePlaceholder}
+                alt="game banner"
+                className={cn(
+                  'absolute inset-0 h-full w-full object-cover transition-opacity duration-300 ease-out',
+                  isImageLoaded ? 'opacity-0' : 'opacity-100',
+                )}
+                style={{
+                  filter: 'blur(6px)',
+                  transform: 'scale(1.1)',
+                  willChange: 'opacity', // immediately prepare the GPU for the transition
+                }}
+                aria-hidden="true"
+              />
+            ) : null}
+
+            {/* Full-resolution image */}
+            <picture
+              className={cn(
+                'absolute inset-0 transition-opacity duration-300 ease-out',
+                isImageLoaded ? 'opacity-100' : 'opacity-0',
+              )}
+              style={{ willChange: 'opacity' }} // immediately prepare the GPU for the transition
+            >
+              <source srcSet={game.banner.mobileSmAvif ?? undefined} type="image/avif" />
+              <source srcSet={game.banner.mobileSmWebp ?? undefined} type="image/webp" />
+
+              {/* Legacy fallback to in-game screenshot */}
+              <img
+                ref={handleImageRef}
+                src={game.imageIngameUrl}
+                alt=""
+                className="h-full w-full object-cover"
+                style={{
+                  objectPosition: isNintendoDS ? 'center 0%' : 'center',
+                }}
+                onLoad={() => setIsImageLoaded(true)}
+                fetchPriority="high"
+                loading="eager"
+                decoding="async"
+              />
+            </picture>
+          </>
+        ) : (
+          // If the game has no custom banner set, fall back to ImageIngame.
```

Comment:
I imagine most games won't have a custom banner as they won't have a big enough fandom to warrant the additional art. Are we expecting a lot of games just to provide upscaled in-game images or boxart crops for their backgrounds?

In that case, I would expect this upscaling behavior to be duplicated for desktop. I know you've said before that you don't want to.

Reply:
I would imagine we're going to have a weird in-between period where most images aren't upscaled for quite some time. However, my sense is the new game page is going to incentivize people to upscale stuff, and a next logical step will be to support higher-quality media using medialibrary. This PR serves as a bit of precursor to that just to prove out we can handle medialibrary functionality at all.

Reply:
> we're going to have a weird in-between period where most images aren't upscaled

> my sense is the new game page is going to incentivize people to upscale stuff

It's something we can discuss more in the desktop support PR, but my concern is that we should still provide a fallback on desktop. I know you don't want to do upscale the in-game image, so what would the fallback be?

Reply:
> We'll just have to be clever in how we do it to make sure the upscaling isn't overly-obvious.

For example: in my original idea for desktop banners, the way I achieved this was making the banner significantly less prominent if it was an upscaled fallback.